unit-tests/utilities/concurrency/test-usage.cpp

```diff
@@ -0,0 +1,194 @@
+// License: Apache 2.0. See LICENSE file in root directory.
```

Comment:
I don't like `test-concurrency-usage`... maybe we should split into `test-scp.cpp` and `test-dispatcher.cpp`?

Reply:
Splitted and more UT added to the dispatcher